code/game/machinery/station_map.dm

```diff
@@ -0,0 +1,321 @@
+var/list/station_holomaps = list()
+
+/obj/machinery/station_map
+	name = "station holomap"
+	desc = "A virtual map of the surrounding station."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "station_map"
+	anchored = 1
+	density = 0
+	use_power = 1
+	idle_power_usage = 10
+	active_power_usage = 10
+	dir = NORTH
+
+	use_auto_lights = 1
+	light_color = "#64C864"
+	light_power_on = 1
+	light_range_on = 2
+
+	machine_flags = SCREWTOGGLE | FIXED2WORK | CROWDESTROY
+
+	layer = ABOVE_WINDOW_LAYER
+
+	var/mob/watching_mob = null
+	var/image/station_map = null
+	var/image/small_station_map = null
+	var/image/floor_markings = null
+	var/image/panel = null
+	var/image/cursor = null
+	var/image/legend = null
+
+	var/original_zLevel = 1
+	var/bogus = 0
```

Comment:
Might want to consider putting a comment for the purpose of this relatively poorly named variable.